tensorboard/http_context.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Context to HTTP request properties."""
+
+
+class HttpContext:
+    """Container of values related to caller's HTTP request.
+
+    This context is for providing security, and quota concerns.
+
+    `HttpContext` values are to be treated as immutable.
+
+    Fields:
+      remote_ip: Best possible guess of the caller's IP Address.
+      x_forwarded_for: Value of X-Forwarded-For header.
```

Comment:
Please document syntax: presumably these are strings representing either
IPv4 or IPv6 addresses? are they known to be in canonical form? could
they also be hostnames, or empty? are they permitted to be `None`? (It
may be that the answers are different for the two fields.)